omisego-sdk/src/main/java/co/omisego/omisego/custom/gson/OMGEnumAdapter.kt

```diff
@@ -8,11 +8,16 @@ package co.omisego.omisego.custom.gson
  */
 
 import co.omisego.omisego.constant.enums.OMGEnum
-import com.google.gson.*
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import com.google.gson.JsonPrimitive
+import com.google.gson.JsonSerializationContext
+import com.google.gson.JsonSerializer
 import java.lang.reflect.Type
 
 @Suppress("UNCHECKED_CAST")
-internal class OMGEnumAdapter<T> : JsonDeserializer<T>, JsonSerializer<T> where T : OMGEnum, T : Enum<T> {
+internal class OMGEnumAdapter<T: OMGEnum> : JsonDeserializer<T>, JsonSerializer<T> {
```

Comment:
Just realized that now that we use `registerTypeHierarchyAdapter`, we only have one instance of `OMGEnumAdapter` common to all enums. 
Meaning that you can't cache anymore like you're doing below in `values`, because you'll cache for the first enum you're de-serializing, and next enums won't create the map since it wont be `null` and thus `values[json.asString]` will return null.

So probably need to replace with

```kotlin
return enumConstants.associateBy { it.value }[json.asString]
```

Reply:
Updated!